### PR TITLE
added homework_4 with 3 files inside

### PR DESCRIPTION
**Pull request by vithaiduchyk**

### CODE REVIEW

File: Homework_8/Task_1.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
+roles = {'admin': 'Vitalii', 'maintainer': 'Josh', 'manager': 'Will', 'developer': 'Steve'}
```
Comment (MarinaZelenska) [inline Homework_8/Task_1.py:2]: values should be lists

File: Homework_8/Task_2.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+
+params = {'v': 'some_id', 'list': 'some_list', 'index': '6', 't': '0s'}
+initial_str = 'https://www.youtube.com/watch?'
+query_string = ''
+
+for key, value in params.items():
+    query_string += f'{key}={value}&'
+
+query_string = query_string[:-1]
```
Comment (MarinaZelenska) [inline Homework_8/Task_2.py:9]: query_string = query_string.strip('&')

Comment (MarinaZelenska) [reply Homework_8/Task_2.py:9]: image

File: Homework_8/Task_3.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+character_counts = {}
+
+for char in result_link:
+    if char in character_counts:
+        character_counts[char] += 1
+    else:
+        character_counts[char] = 1
+
+print(f'Here is result: {character_counts}')
```
Comment (MarinaZelenska) [inline Homework_8/Task_3.py:11]: image

**Review by MarinaZelenska — APPROVED** [review]

File: Homework_8/Task_1.py (hunk 1 above)
Comment (vithaiduchyk) [reply Homework_8/Task_1.py:2]: is it should be couple names for one key?

Comment (vithaiduchyk) [reply Homework_8/Task_1.py:2]: Fixed it and added

File: Homework_8/Task_2.py (hunk 2 above)
Comment (vithaiduchyk) [reply Homework_8/Task_2.py:9]: fixed and added to new pull request

**Pull request closed without merging**